services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,131 @@
+package cromwell.services.healthmonitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Scheduler, Timers}
+import akka.pattern.{after, pipe}
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.language.postfixOps
+import HealthMonitorServiceActor._
+import com.typesafe.config.Config
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import net.ceedubs.ficus.Ficus._
+
+/**
+  * A trait which provides a health monitoring service. Implementers are expected to provide a collection of functions
+  * which will each check some underlying service on which Cromwell depends, for instance connectivity to the database
+  * or DockerHub. At a regular interval the health monitor actor will run each of these functions and keep their
+  * current state in a cache.
+  *
+  * This actor may also be interrogated for the current status of the monitored subsystems. The response will indicate
+  * if all of the subsystems are deemed to be healthy, and if this is not true subsystem-specific messages will be
+  * returned to the client.
+  */
+trait HealthMonitorServiceActor extends Actor with LazyLogging with Timers {
+  val serviceConfig: Config
+  def subsystems: Set[MonitoredSubsystem]
+
+  implicit val ec: ExecutionContext = context.system.dispatcher
+
+  lazy val futureTimeout: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-timeout").getOrElse(DefaultFutureTimeout)
+  lazy val staleThreshold: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.status-ttl").getOrElse(DefaultStaleThreshold)
+
+  override def preStart(): Unit = {
+    val sweepTime = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-refresh-time").getOrElse(DefaultSweepTime)
```

Comment:
as above. It'd be nice to take out `serviceConfig` and either declare some kind of config class or just declare each field as abstract.

Reply:
Talked about in person, but `serviceConfig` is part of the contract of the service registry (whether that's a good idea or not is debatable but I'd like to think of it as out of scope here)